docs/sphinx/user_guide/feature/array.rst

```diff
@@ -0,0 +1,170 @@
+
+.. _array-label:
+
+======
+Arrays
+======
+
+Camp has its own Array implementation very similar to C++ std::array. Array functionality is
+included in the ``array.hpp`` file within ``include/camp/``. 
+
+Some important notes: 
+
+.. note:: All Camp structures and classes are in the namespace ``camp``.
+          
+----------------
+Array Philosophy
+----------------
+
+In C++, an array is a constant-sized, statically allocated slice of contiguous memory. 
+While C++ arrays are conceptually equivalent to C-style arrays--in that their size must be known 
+at compile time--they have many quality-of-life features that make them easier and more convenient to use. 
+
+For example, in C, one might allocate memory in as follows:: 
```

Comment:
Would it be better to condense the C and C++ array discussion to something like this:

double arr[N];

This is valid in C and C++ and illustrates a static (i.e., automatic) allocation.

The examples using malloc and new are run time allocations from the free store, which is not the same as the std::array example below. That is also a static allocation.

Reply:
careful of terms here statis storage duration is for storage that lasts the length of the program, automatic lasts until it goes out of scope, and dynamic is user controlled.